packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -427,6 +444,37 @@ const MetricForm: FC<MetricFormProps> = ({
     }
   }, [type, form]);
 
+  const tableOptions: { label: string; value: string }[] = [];
+
+  const { data: TableData } = useApi<{
+    informationSchema: InformationSchemaInterface;
+  }>(`/datasource/${selectedDataSource.id}/schema`);
```

Comment:
This will make an API call even if the data source doesn't support information schemas.